### PR TITLE
SOC-3838 | X-Trace-Id for backend auth calls

### PR DESCRIPTION
**Pull request by bkoval**
## Links

* http://wikia-inc.atlassian.net/browse/SOC-3838

## Reviewers

@macbre @Wikia/social-backend

### CODE REVIEW

File: server/app/lib/wikia-session.js [hunk 1]
```diff
@@ -61,7 +62,8 @@ export default function scheme() {
 				{
 					headers: {
 						'X-Client-Ip': request.headers['fastly-client-ip'] || request.info.remoteAddress,
-						'X-Forwarded-For': request.headers['x-forwarded-for'] || request.info.remoteAddress
+						'X-Forwarded-For': request.headers['x-forwarded-for'] || request.info.remoteAddress,
+						'X-Trace-Id': request.headers['x-trace-id'] || uuid.v4()
```
Comment (macbre) [inline server/app/lib/wikia-session.js]: Copy&paste detected here and in files above :) Can we introduce a helper that will return all the required headers?

File: server/app/facets/operations/username.js [hunk 2]
```diff
@@ -16,7 +17,8 @@ function createUserRegistrationContext(services, data, request) {
 			headers: {
 				'X-Wikia-Internal-Request': 1,
```
Comment (macbre) [inline server/app/facets/operations/username.js]: Please set `X-Wikia-Internal-Request` to a more descriptive value (e.g. `mercury`). We log these values from MediaWiki and it's sometimes helpful when debugging performance issues.

File: server/app/lib/utils.js [hunk 3]
```diff
@@ -342,3 +344,21 @@ export function setI18nLang(request, wikiVariables) {
...
+ * @param {Object} customHeaders
+ * @returns {Object}
+ */
+export function getInternalHeaders(request, customHeaders = {}) {
+	return deepExtend(
+		{},
+		{
+			'X-Client-Ip': request.headers['fastly-client-ip'] || request.info.remoteAddress,
+			'X-Forwarded-For': request.headers['x-forwarded-for'] || request.info.remoteAddress,
+			'X-Trace-Id': request.headers['x-trace-id'] || uuid.v4()
```
Comment (slayful) [inline server/app/lib/utils.js]: 👍

File: server/app/facets/operations/validate-token.js [hunk 4]
```diff
@@ -4,16 +4,15 @@ import Promise from 'bluebird';
 import settings from '../../../config/settings';
 import Wreck from 'wreck';
 import translateUserIdFrom from './username';
+import {getInternalHeaders} from '../../lib/utils';
```
Comment (slayful) [inline server/app/facets/operations/validate-token.js]: 👍

**Comment by slayful** [comment]
LGTM

**Pull request merged**